tests/phpunit/tests/media.php

```diff
@@ -3798,6 +3798,64 @@ function() use ( &$result ) {
 
 		$this->assertFalse( $result );
 	}
+
+	/**
+	 * @ticket 55443
+	 */
+	public function test_wp_wepb_fallback_if_content_has_updated_images() {
+
+		if ( ! wp_image_editor_supports( array( 'mime_type' => 'image/webp' ) ) ) {
+			$this->markTestSkipped( 'This test requires WebP support.' );
+		}
+
+		remove_all_actions( 'template_redirect' );
+		do_action( 'template_redirect' );
+
+		$filename      = DIR_TESTDATA . '/images/' . self::$large_filename;
+		$attachment_id = $this->factory->attachment->create_upload_object( $filename );
+
+		// Preferring WebP over JPEG results in no changes to content.
+		add_filter(
+			'wp_content_image_mimes',
+			function() {
+				return array( 'image/webp', 'image/jpeg' );
+			}
+		);
+
+		apply_filters(
+			'the_content',
+			sprintf(
+				'<p>before image</p>%s<p>after image</p>',
+				wp_filter_content_tags( wp_get_attachment_image( $attachment_id, 'medium', false, array( 'class' => "wp-image-{$attachment_id}" ) ) )
+			)
+		);
+
+		$this->assertTrue( has_action( 'wp_footer', 'wp_wepb_fallback' ) === 10 );
+
+		ob_start();
+			wp_footer();
+		$footer = ob_get_clean();
+		$this->assertStringContainsString( 'data:image/webp;base64,UklGR', $footer );
+	}
+
+	/**
+	 * @ticket 55443
+	 */
+	public function test_wp_wepb_fallback_if_content_has_no_updated_images() {
+
+		if ( ! wp_image_editor_supports( array( 'mime_type' => 'image/webp' ) ) ) {
+			$this->markTestSkipped( 'This test requires WebP support.' );
+		}
```

Comment:
do we need to check for image editor support for WebP here? I would say we should put some more focus on removing these checks for each tests and having some mock image editor that supports WebP. We have this in performance plugin and we might be able use it here. https://github.com/WordPress/performance/blob/525eeb859f34a3bc4dbfac9444bfa62ac6006da0/tests/modules/images/webp-uploads/load-tests.php#L583

Reply:
the core tests run against a variety of environments, some with GD, some Imagick and some that don't support WebP. Because this test requires WebP support, we skip it for this run. Does that make sense? other core image tests work this way.

Reply:
Yes make sense. In future if we find any better solution for this then we will update this condition.